rotkehlchen/tests/unit/test_ethereum_names.py

```diff
@@ -0,0 +1,73 @@
+from typing import Dict, List
+from unittest.mock import Mock
+
+import pytest
+
+from rotkehlchen.chain.ethereum.names import FetcherFunc, NamePrioritizer
+from rotkehlchen.tests.utils.factories import make_ethereum_address
+from rotkehlchen.types import DEFAULT_ADDRESS_NAME_PRIORITY, AddressNameSource, ChecksumEvmAddress
+
+
+@pytest.fixture(name='evm_address')
+def fixture_evm_address() -> ChecksumEvmAddress:
+    return make_ethereum_address()
+
+
+@pytest.fixture(name='name_source_ids')
+def fixture_name_source_ids() -> List[AddressNameSource]:
+    return DEFAULT_ADDRESS_NAME_PRIORITY
+
+
+@pytest.fixture(name='fetchers')
+def fixture_fetchers(
+        name_source_ids,
+) -> Dict[AddressNameSource, FetcherFunc]:
+    fetchers: Dict[AddressNameSource, FetcherFunc] = {}
+    for source_id in name_source_ids:
+        def make_fetcher(src) -> FetcherFunc:
+            return lambda db, addr: f"{src} address name"
+
+        fetchers[source_id] = make_fetcher(source_id)
+    return fetchers
+
+
+@pytest.fixture(name='name_sources_with_lowest_prio_has_name')
+def fixture_name_sources_with_lowest_prio_has_name(
+        name_source_ids,
+        fetchers,
+) -> Dict[AddressNameSource, FetcherFunc]:
+    def empty_source(_db, _addr):
+        return ""
+
+    for i, name_source_id in enumerate(name_source_ids):
+        if i == len(name_source_ids) - 1:  # do nothing with lowest prio name source
+            break
+
+        fetchers[name_source_id] = empty_source
+    return fetchers
+
+
+def test_get_prioritized_name(name_source_ids, fetchers, evm_address):
```

Comment:
Write a docstring here and in the next test explaining what you are testing. It's a small test but very hard to follow

Reply:
I'll rewrite the tests to be easier to follow.